include/cadmium/modeling/dynamic_models_helpers.hpp

```diff
@@ -32,109 +32,179 @@
-         *
-         * @tparam BST The message bag tuple to fill from the dynamic_message_bags.
-         * @param bags - The dynamic_message_bags that carries the message to be placed in the bs parameter.
-         * @param bs  - The BST message bags that will be filled with the bags messages.
-         */
-        template<typename BST>
-        void fill_bags_from_map(cadmium::dynamic_message_bags& bags, BST& bs) {
-
-            auto add_messages_to_bag = [&bags](auto & b)->void {
-                using bag_type = decltype(b);
-
-                bag_type b2 = boost::any_cast<bag_type>(bags.at(typeid(bag_type)));
-                b.messages.insert(b.messages.end(), b2.messages.begin(), b2.messages.end());
-            };
-            for_each<BST>(bs, add_messages_to_bag);
-        }
-
-        /**
-         * @brief Insert all the message bs of bags in bags by an implicit conversion of them to boost::any.
-         *
-         * @tparam BST The message bag tuple that carries the messages to fill the dynamic_message_bags.
-         * @param bags - The dynamic_message_bag that will be filled with the bs messages.
-         * @param bs  - The BST message bag that carries the message to be placed in the bags parameter.
-         */
-        template<typename BST>
-        void fill_map_from_bags(BST& bs, cadmium::dynamic_message_bags& bags) {
-
-            auto add_messages_to_map = [&bags](auto & b)->void {
-                using bag_type = decltype(b);
-
-                bags[typeid(bag_type)] = b;
-            };
-            for_each<BST>(bs, add_messages_to_map);
-        }
-
-        bool is_in(const std::type_index& port, const ports_vector& ports) {
-            return std::find(ports.cbegin(), ports.cend(), port) != ports.cend();
-        }
-
-        bool valid_ic_links(const models_map& models, const IC_vector& ic) {
-            return std::all_of(ic.cbegin(), ic.cend(), [&models](const auto& link) -> bool {
-                return models.find(std::get<0>(link)) == models.cend() || models.find(std::get<2>(link)) == models.cend();
-            });
-        }
-
-        bool valid_eic_links(const models_map& models, const ports_vector& input_ports, const EC_vector& eic) {
-            return std::all_of(eic.cbegin(), eic.cend(), [&models, &input_ports](const auto& link) -> bool {
-                return models.find(std::get<1>(link)) == models.cend() || is_in(std::get<0>(link), input_ports);
-            });
-        }
-
-        bool valid_eoc_links(const models_map& models, const ports_vector& output_ports, const EC_vector& eoc) {
-
-            return std::all_of(eoc.cbegin(), eoc.cend(), [&models, &output_ports](const auto& link) -> bool {
-                return models.find(std::get<0>(link)) == models.cend() || is_in(std::get<2>(link), output_ports);
-            });
+    namespace dynamic {
+        namespace modeling {
+
+            // Generic tuple for_each function
```

Comment:
Do we have this one double defined? Here and in the modeling one? If so, we should extract. It could become a problem if including both headers somehow.